packages/lib/auth.ts

```diff
@@ -32,5 +32,8 @@ export function isPasswordValid(password: string) {
       if (password[i] === password[i].toLowerCase()) low = true;
     }
   }
-  return cap && low && num && min;
-}
+  return !!breakdown ? { caplow: cap && low, num, min } : cap && low && num && min;
+}) as {
+  (password: string): boolean;
+  (password: string, breakdown: boolean): { caplow: boolean; num: boolean; min: boolean };
+};
```

Comment:
This is using function overloads to make it clear that when providing just the `password` string, you will be getting a boolean in result, or if you provide a truthy second parameter, you get the breakdown of the checks for the password.

Reply:
Can we just declare them on top instead of casting?

```ts
export function isPasswordValid(password: string): boolean;
export function isPasswordValid(password: string, breakdown: boolean): { caplow: boolean; num: boolean; min: boolean };
export function isPasswordValid(password: string) {
```

Reply:
Of course, my bad. Here it is: https://github.com/calcom/cal.com/pull/3598